pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1854,23 +1859,63 @@ private void setRowBufferColumn(Tuple rowBuffer,
         // or we won't be able to re-parse it.
         //
         case Types.DATE:
-          rowBuffer.set(columnIndex, connection
-              .encodeString(
-                  connection.getTimestampUtils().toString(
-                      getDefaultCalendar(), (Date) valueObject)));
+          if (valueObject instanceof Timestamp) {
+            rowBuffer.set(columnIndex, connection
+                .encodeString(
+                    connection.getTimestampUtils().toString(
+                        getDefaultCalendar(), (Date) valueObject)));
+
+          } else if (valueObject instanceof LocalDate) {
+            rowBuffer.set(columnIndex, connection
```

Comment:
Why aren't you using `org.postgresql.jdbc.TimestampUtils#toString(LocalDate)` here?

Reply:
Thanks, fixed.